src/content/en/2021/javascript.md

```diff
@@ -879,6 +890,8 @@ The numbers that we have seen throughout the chapter have brought us to an under
 
 It is observed that so many features provided for improving the rendering time and resource loading time could be leveraged better to see the overall impact and get an even better experience with respect to the performance.
 
-Starting from adopting the features provided by the JavaScript/Web community to using them wisely and ensuring that a tweak only improves the performance and doesn't instead increase the overload (for example, lazy loading all resources on a page is definitely not a good idea), is what it will take us to see these numbers even better in the coming years.
+Starting from adopting the features provided by the JavaScript and web community, making sure to use them wisely and ensuring that a tweak only improves the performance and doesn't instead increase the overload (for example, lazy loading all resources on a page is definitely not a good idea), is what it will take us to see these numbers even better in the coming years.
+
+{# TODO - this sentence is long but not sure how to break it down #}
```

Comment:
Agreed and not sure that this is trying to say. Does this cover it?

> Start by adopting the new features provided by the JavaScript and web community. However, make sure to use them wisely and ensure that they improve performance and don't instead increase the load, as some APIs can cause harm through overused. For example, lazy loading should be applied selectively and not bluntly set on all resources on a page. Making appropriate use of the powerful APIs we now have access to, is what it will take to see these numbers improve further in the coming years.

FYI @shantsis if you select all three lines when making comments than people can suggest fixes in these comments and commit straight away from these comments. At the moment I can only suggest changes to the line you've commented on (the TODO bit).

Reply:
Yup this covers what I was trying to say. 
"through overused"? - do you mean "through overuse"?

Reply:
Yup. Edited it above.

Is lazy loading the best example though? Especially with native lazy loading now so common, using JS for this is less necessary now (I argue not necessary at all!). Wonder if there's a better example we could use here that's mroe relatable to JS?

Reply:
thanks for the tip @tunetheweb!

Reply:
>Wonder if there's a better example we could use here that's more relatable to JS?

How about async/defer?
defer 's usage becoming meaningless in the presence of async.

Reply:
Better example!

> Start by adopting the new features provided by JavaScript and the web community. However, make sure to use them wisely and ensure that they improve performance and don't instead increase the load, as some APIs can cause harm through overuse. For example, we saw that using `async` as well as `defer` is very common and actually is an anti-pattern that slows the site down since the `defer` is effectively ignored. Making appropriate use of the powerful APIs we now have access to, is what it will take to see these numbers improve further in the coming years.